proteuslib/tools/parameter_sweep.py

```diff
@@ -129,9 +211,29 @@ def _default_optimize(model, options=None, tee=False):
 
 # ================================================================
 
+def _process_sweep_params(sweep_params):
+
+    # Check the list of parameters to make sure they are valid
+    for i, key in enumerate(sweep_params.keys()):
+
+        # Convert to using Sample class
+        if isinstance(sweep_params[key],(list,tuple)):
+            sweep_params[key] = LinearSample(*sweep_params[key])
+
+        # Get the type of sampling
+        if i == 0:
+            sampling_type = getattr(sweep_params[key],"type",None)
+        else:
+            if sampling_type != getattr(sweep_params[key],"type",None):
+                raise ValueError("Cannot mix sampling types")
```

Comment:
Here's where we could just check `isinstance(sweep_param[key], RandomSample)` or such if we used stronger typing.

Reply:
I updated this logic to take advantage of the new FixedSample and RandomSample classes